OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs

```diff
@@ -25,6 +25,36 @@ protected Actor FindClosestEnemy(Squad owner)
 		{
 			return owner.SquadManager.FindClosestEnemy(owner.Units.First().CenterPosition);
 		}
+
+		protected Actor GetRandomAttackable(Squad owner)
+		{
+			var manager = owner.SquadManager;
+			var allEnemyBaseBuilder = AIUtils.FindEnemiesByCommonName(owner.SquadManager.Info.ConstructionYardTypes, owner.SquadManager.Player);
+
+			if (!allEnemyBaseBuilder.Any())
+				return FindClosestEnemy(owner);
+
+			foreach (var b in allEnemyBaseBuilder)
+			{
+				var enemies = manager.World.FindActorsInCircle(b.CenterPosition, WDist.FromCells(manager.Info.RushAttackScanRadius))
+					.Where(u => ((manager.IsEnemyUnit(u) && manager.IsNotHiddenUnit(u)) || manager.IsEnemyUnit(u)) && u.Info.HasTraitInfo<MustBeDestroyedInfo>()).ToList();
```

Comment:
Due to the `|| manager.IsEnemyUnit(u)`, the `(manager.IsEnemyUnit(u) && manager.IsNotHiddenUnit(u))` before it doesn't make a difference, so one of these should be dropped.

Reply:
`.Where(u => manager.IsEnemyUnit(u) && u.Info.HasTraitInfo<MustBeDestroyedInfo>()).ToList();`
is equivalent to the current expression, so that can be used.

Reply:
The question is if hidden units should really be included.

Reply:
Hi here.
I just wanted to do some check that picked out unhidden one first, then the hiden one if not any unhidden one, I implemented it in a wrong way, though. Thanks for point out!

Reply:
fixed

Reply:
Emmm wait, I need to rewrite those lines

Reply:
Done. I comprehended the lines in `TryToRushAttack` wrong at some places.